Extensions/BitmapText/JsExtension.js

```diff
@@ -0,0 +1,588 @@
+        .getOrCreate('color')
+        .setValue(objectContent.color)
+        .setType('color')
+        .setLabel(_('Base color'));
+
+      objectProperties
+        .getOrCreate('opacity')
+        .setValue(objectContent.opacity.toString())
+        .setType('number')
+        .setLabel(_('Opacity (0-255)'));
+
+      objectProperties
+        .getOrCreate('fontSize')
+        .setValue(objectContent.fontSize.toString())
+        .setType('number')
+        .setLabel(_('Base size'));
+
+      objectProperties
+        .getOrCreate('align')
+        .setValue(objectContent.align)
+        .setType('choice')
+        .addExtraInfo('left')
+        .addExtraInfo('center')
+        .addExtraInfo('right')
+        .setLabel(_('Base alignment'));
+
+      objectProperties
+        .getOrCreate('fontFamily')
+        .setValue(objectContent.fontFamily)
+        .setType('resource')
+        .addExtraInfo('font')
+        .setLabel(_('Base font family'));
+
+      objectProperties
+        .getOrCreate('wordWrap')
+        .setValue(objectContent.wordWrap ? 'true' : 'false')
+        .setType('boolean')
+        .setLabel(_('Word wrapping'));
+
+      objectProperties
+        .getOrCreate('visible')
+        .setValue(objectContent.visible ? 'true' : 'false')
+        .setType('boolean')
+        .setLabel(_('Visible on start'));
+
+      return objectProperties;
+    };
+    objectBitmapText.setRawJSONContent(
+      JSON.stringify({
+        text: 'Bitmap Text test',
+        opacity: 255,
+        fontSize: 20,
+        visible: true,
+        color: '#000000',
+        fontFamily: 'Arial',
+        align: 'left',
+        wordWrap: true,
+      })
+    );
+
+    // $FlowExpectedError
+    objectBitmapText.updateInitialInstanceProperty = function(
+      objectContent,
+      instance,
+      propertyName,
+      newValue,
+      project,
+      layout
+    ) {
+      return false;
+    };
+    // $FlowExpectedError
+    objectBitmapText.getInitialInstanceProperties = function(
+      content,
+      instance,
+      project,
+      layout
+    ) {
+      var instanceProperties = new gd.MapStringPropertyDescriptor();
+      return instanceProperties;
+    };
+
+    const object = extension
+      .addObject(
+        'BitmapText',
+        _('BitmapText'),
+        _(
+          'Displays a rich text label using Bitmap markup (allowing to set parts of the text as bold, italic, use different colors and shadows).'
+        ),
+        'JsPlatform/Extensions/bitmapfont32.png',
+        objectBitmapText
+      )
+      .setIncludeFile('Extensions/BitmapText/bitmaptextruntimeobject.js')
+      .addIncludeFile(
+        'Extensions/BitmapText/bitmaptextruntimeobject-pixi-renderer.js'
+      );
+
+    /**
+     * Utility function to add both a setter and a getter to a property from a list.
+     * Useful for setting multiple generic properties.
+     */
+    const addSettersAndGettersToObject = (gdObject, properties, objectName) => {
```

Comment:
Note that for this extension, I would prefer that we use the traditional addAction/addCondition. We can change that later :)